plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintExtension.kt

```diff
@@ -61,4 +61,12 @@ open class KtlintExtension(
     val reporters: SetProperty<ReporterType> = objectFactory.setProperty {
         set(setOf(ReporterType.PLAIN))
     }
+
+    /**
+     * Exclude sources under given source patterns.
+     *
+     * See [PatternFilterable](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html)
+     * for details how source patterns should look.
+     */
+    val exclude: SetProperty<String> = objectFactory.setProperty()
```

Comment:
I'd rather you just expose a method `fun matching( filterConfigAction: Action<PatternFilterable>)` instead.
That way any consumer can specify a custom matching pattern for inclusion/exclusion at their leisure.

Reply:
Do I understand it right - you want to have in extension something like this?

```kotlin
    internal var filter: Action<PatternFilterable> = Action {} 
    fun matching(filter: Action<PatternFilterable>) {
        this.filter = filter
    }
```

Reply:
I think that makes sense. Do you?

Reply:
I am a little worry about allowing to use `include()` and possible misunderstanding from users, as they may think that they should include manually all kotlin sources.

On the other hand I see a benefit as well of such approach.

Reply:
They can do the same thing if they use `tasks.withType<KtlintCheckTask>().configureEach {`

Reply:
valid point :+1: